underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AInputStream.java

```diff
@@ -153,15 +152,9 @@ private void openStream() throws IOException {
         errorMessage = String
             .format("Failed to open key: %s bucket: %s attempts: %d error: %s", mKey, mBucketName,
                 mRetryPolicy.getAttemptCount(), e.getMessage());
-        if (e.getStatusCode() != HttpStatus.SC_NOT_FOUND) {
-          throw AlluxioS3Exception.from(errorMessage, e);
-        }
-        // Key does not exist
-        lastException = e;
+        throw AlluxioS3Exception.from(errorMessage, e);
```

Comment:
Could you elaborate on the reason why we change the behavior here? seems like inputStream would not be covered by the ObjectUnderFileSystem retry @Jackson-Wang-7

Reply:
Actually, if ufs inputStream meet exceptions, `alluxio.client.file.AlluxioFileInStream#read()` will rebuild the stream and read again until it reaches max duration time. and after the discussion with @yyongycy and @yuzhu, I don't think we need to retry it here for the NotFound error code anymore.

Reply:
That makes sense, thanks for the explaintion. Just another thought that I think it would be better to handle retry specific exception in the ufs here so we at higher level we don't blindly retry all exceptions.